tests/baselines/reference/privateConstructorJsDocDeclarationEmit.js

```diff
@@ -0,0 +1,82 @@
+//// [tests/cases/conformance/salsa/privateConstructorJsDocDeclarationEmit.ts] ////
+
+//// [privateConstructorJsDocDeclarationEmit.js]
+// https://github.com/microsoft/TypeScript/issues/58653
+
+class Foo {
+    /**
+     * @private
+     */
+    constructor() { }
+}
+
+class Bar {
+    constructor() { }
+}
+
+class Baz extends Bar {
+    /**
+     * @private
+     */
+    constructor() {
+        super()
+    }
+}
+
+// https://github.com/microsoft/TypeScript/pull/58655#issuecomment-2130552519
+
+class C {
+  /**
+   * @private
+   * @overload
+   * @param x {string}
+   * @param y {string}
+   */
+
+  /**
+   * @private
+   * @overload
+   * @param x {number}
+   * @param y {number}
+   */
+
+  /**
+   * @private
+   * @param {...any} args
+   */
+  constructor(...args) {}
+}
+
+
+
+//// [privateConstructorJsDocDeclarationEmit.d.ts]
+declare class Foo {
+    /**
+     * @private
+     */
+    private constructor();
+}
+declare class Bar {
+}
+declare class Baz extends Bar {
+    /**
+     * @private
+     */
+    private constructor();
+}
+declare class C {
+    /**
+     * @private
+     * @overload
+     * @param x {string}
+     * @param y {string}
+     */
+    constructor(x: string, y: string);

```

Comment:
This is not correct because we shouldnt be emitting types for private - this could reference private type

Reply:
yes, this is invalid but it's a different bug - this test case has been suggested by @DanielRosenwasser and I'm investigating now (separately) what can be done to fix this.

The problem is not about what gets emitted here but about the fact that it's not recognized as private at all. Notice how there is no error here: [TS playground](https://www.typescriptlang.org/play/?ts=5.5.0-dev.20240525&filetype=js#code/MYGwhgzhAEDC0G8BQ1XQPQCpMrdT0AAgA4BOAlgG5gAuAprmgYQPaV2kgtgAmjqzYmFJgAttAAeiCDQoA7AOYBffviJCR4gJ7TZ5RSrz50SflhxHBFavVXM2HLrzvrhYyYjkBXUQCMOhniCbtqePv6kgUwmZtguJNa0DJaumogAdJlgclpK0MIKMHYmeMAscjKkXsA0LKQAFJnpBRAAlIimeCoqSHJ0AO5w9QCMADTQw62m06ZAA)